core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -263,9 +264,11 @@ public void renameTable(SessionContext context, TableIdentifier from, TableIdent
     client.post(paths.rename(), request, null, headers(context), ErrorHandlers.tableErrorHandler());
   }
 
-  private LoadTableResponse loadInternal(SessionContext context, TableIdentifier identifier) {
+  private LoadTableResponse loadInternal(
+      SessionContext context, TableIdentifier identifier, SnapshotLoading snapshotLoading) {
```

Comment:
I think it would be much simpler to make this a catalog option that is set as a boolean field or a local enum. That would require no changes to this method definition and wouldn't require changing paths at all.

Reply:
While we can control this at a catalog level (i.e. to set the default behavior), but we need to change the method signature so that we can [override the behavior when lazy loading](https://github.com/apache/iceberg/pull/6850/files#diff-86450612dbe323d6d06cbc3846aa1913f042eaedadc0ca027c36bfbe08d3a46cR315).  We would either need a new method to load refs/all or pass in the option like this.

Reply:
Makes sense to pass this through as an arg then, but I'd definitely prefer not making changes to `ResourcePaths`